### PR TITLE
Additional User RP Tests

### LINKED ISSUE #244: Missing User Risk Premium Tests
Missing User Risk Premium Tests:

* user risk premium decreases after repayment
* user risk premium increases after withdrawal (or removal of collateral)
* user risk premium changes because of collateral accrual (no debt change)
* user risk premium changes because of debt accrual (no collateral change)
* user risk premium changes because of coll and debt accrual

### PR DESCRIPTION
**Pull request by CheyenneAtapour**
closes #244

### CODE REVIEW

File: src/contracts/Spoke.sol [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import {console2 as console} from 'forge-std/console2.sol';
```
Comment (yan-man) [inline src/contracts/Spoke.sol]: rm

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+import 'tests/unit/Spoke/SpokeBase.t.sol';
+
+contract SpokeRiskPremiumEdgeCasesTest is SpokeBase {
+  using SharesMath for uint256;
+  using WadRayMath for uint256;
+  using PercentageMath for uint256;
+
+  /// Bob supplies 2 collateral assets, borrows an amount such that both of them cover it, and then repays any nonzero amount of the higher lp one
+  /// Bob's user risk premium should decrease or remain same after repay
+  /// @dev due to rounding within risk premium calc, repaying doesn't guarantee user rp decrease
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:12]: is it only the repay donation scenario that ends up with the rp not decreasing?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+  function test_riskPremium_nonIncreasingAfterRepay(
+    uint256 usdxSupplyAmount,
+    uint256 daiSupplyAmount,
+    uint256 borrowAmount,
+    uint256 repayAmount
+  ) public {
+    // Make usdx liquidity premium 10 so it's the lower lp asset
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]:
```suggestion
    // Make usdx liquidity premium 10% so it's the lower lp asset
```

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+    updateLiquidityPremium(spoke2, _usdxReserveId(spoke2), 10_00);
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:20]: can perhaps add an assert that usdx is indeed the lower lp asset

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+pragma solidity ^0.8.0;
+
+import 'tests/unit/Spoke/SpokeBase.t.sol';
+
+contract SpokeRiskPremiumEdgeCasesTest is SpokeBase {
+  using SharesMath for uint256;
+  using WadRayMath for uint256;
+  using PercentageMath for uint256;
+
+  /// Bob supplies 2 collateral assets, borrows an amount such that both of them cover it, and then repays any nonzero amount of the higher lp one
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: is this meant to be 3 coll assets then? curious why you used dai/dai2 and 3 coll assets instead of 2

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+      spoke: spoke2,
+      user: bob,
+      reserveId: _daiReserveId(spoke2),
+      usingAsCollateral: false
+    });
+
+    assertGt(
+      spoke2.getUserRiskPremium(bob),
+      riskPremium,
+      'Risk premium should increase after removing collateral'
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]:
```suggestion
      'Risk premium should increase after disabling lower LP asset as collateral'
```

or something like this

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+      reserveId: _daiReserveId(spoke2),
+      user: bob,
+      amount: daiSupplyAmount,
+      onBehalfOf: bob
+    });
+
+    assertGt(
+      spoke2.getUserRiskPremium(bob),
+      riskPremium,
+      'Risk premium should increase after withdrawing collateral'
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]:
```suggestion
      'Risk premium should increase after withdrawing lower LP collateral'
```

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 5 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: dai2 is just used for borrowing power, and it's higher LP than the other 2 collaterals being tested in the user RP here, so it doesn't affect the user rp calc, intentionally

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 2 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:12]: No, it can be due to rounding in general because we are using bps values. 1 wei of asset difference won't generally change the user rp calc

File: src/contracts/Spoke.sol (hunk 1 above)
Comment (CheyenneAtapour) [reply src/contracts/Spoke.sol]: done, ty

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 4 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:20]: done

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 3 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: done

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 6 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: done

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 7 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: done

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+
+    daiSupplyAmount = bound(daiSupplyAmount, 4, MAX_SUPPLY_AMOUNT);
+    borrowAmount = bound(borrowAmount, 3, MAX_SUPPLY_AMOUNT / 2);
+    // Force least lp asset supply amount to be less than borrow amount, so borrow covered by 2 collaterals at least
+    // Here it will be due to decimals of usdx vs dai
+    usdxSupplyAmount = bound(usdxSupplyAmount, 1, borrowAmount);
+    repayAmount = bound(repayAmount, 2, borrowAmount);
+
+    // Deal bob dai to cover dai and dai2 supply
+    deal(address(tokenList.dai), bob, MAX_SUPPLY_AMOUNT * 2);
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:38]: maybe can add a setup method and include this deal in there, ig only used 2 times nbd

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+    // Now bob withdraws dai
+    Utils.withdraw({
+      spoke: spoke2,
+      reserveId: _daiReserveId(spoke2),
+      user: bob,
+      amount: daiSupplyAmount,
+      onBehalfOf: bob
+    });
+
+    assertGt(
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: can it also be Ge for the same rationale, ie if some dust amount is withdrawn? maybe can add a fuzz test where not full `daiSupplyAmount`, but just some fuzzed amount <= `daiSupplyAmount` is withdrawn

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:1]: nice test scenarios. Maybe also:
- some fuzzed versions of these
- initially 1 coll covers debt, then due to borrowing more of another asset (or same asset), it changes RP to be covered by 2 colls
- initially 1 coll covers debt, then adding lower lp coll changes (or completely replaces) rp 

apologies if this was already incl in the previous pr, i forget

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 8 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:38]: Yeah, it's just a niche use-case here, I think fine to leave as is

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 9 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: Yes, that's right. I'll write a fuzz test with that

Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: Added

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 10]
```diff
@@ -0,0 +1,1048 @@
...
+    });
+
+    // Bob's current risk premium should be greater than liquidity premium of wbtc, since debt is not fully covered by it
+    assertGt(
+      spoke1.getUserRiskPremium(bob),
+      spoke1.getLiquidityPremium(_wbtcReserveId(spoke1)),
+      'Bob user rp after borrow'
+    );
+
+    // Mock calls to ensure dai debt does not does not grow due to interest
```
Comment (DhairyaSethi) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: kinda funky workaround, how about instead we make wbtc's index so high it grows faster than the debt and then eventually is enough to cover entire debt?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 11]
```diff
@@ -0,0 +1,1048 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import 'tests/unit/Spoke/SpokeBase.t.sol';
+
+contract SpokeRiskPremiumEdgeCasesTest is SpokeBase {
+  using SharesMath for uint256;
+  using WadRayMath for uint256;
+  using PercentageMath for uint256;
```
Comment (miguelmtzinf) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: probably unneeded

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 12]
```diff
@@ -0,0 +1,1048 @@
...
+    updateLiquidityPremium(spoke2, _usdxReserveId(spoke2), 10_00);
+    assertLt(
+      spoke2.getLiquidityPremium(_usdxReserveId(spoke2)),
+      spoke2.getLiquidityPremium(_daiReserveId(spoke2)),
+      'Usdx lower lp than dai'
+    );
+
+    daiSupplyAmount = bound(daiSupplyAmount, 4, MAX_SUPPLY_AMOUNT);
+    borrowAmount = bound(borrowAmount, 3, MAX_SUPPLY_AMOUNT / 2);
+    // Force least lp asset supply amount to be less than borrow amount, so borrow covered by 2 collaterals at least
```
Comment (miguelmtzinf) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:29]: does it make sense to dynamically calculate supply amounts based on borrow amount to make sure any of the two collateral assets can cover the debt?
if we change the price of this assets in the setup this won't work

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 13]
```diff
@@ -0,0 +1,1048 @@
...
+    // Here it will be due to decimals of usdx vs dai
+    usdxSupplyAmount = bound(usdxSupplyAmount, 1, borrowAmount);
+    repayAmount = bound(repayAmount, 2, borrowAmount);
+
+    // Deal bob dai to cover dai and dai2 supply
+    deal(address(tokenList.dai), bob, MAX_SUPPLY_AMOUNT * 2);
+
+    // Supply max dai2, the highest lp asset, to allow borrowing without affecting RP
```
Comment (miguelmtzinf) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:40]: why do we need this if the other collateral assets are enough to cover the debt? it's because they can be greater than the debt but not enough for the HF?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 9 above)
Comment (miguelmtzinf) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: cannot we merge the 2 of them into 1? with gte

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 14]
```diff
@@ -0,0 +1,1048 @@
...
+    });
+
+    assertGt(
+      spoke2.getUserRiskPremium(bob),
+      riskPremium,
+      'Risk premium should increase after disabling lower LP reserve as collateral'
+    );
+  }
+
+  /// Supply dai2 and dai as collateral, borrow dai2, then withdraw dai as collateral and risk premium should increase
```
Comment (miguelmtzinf) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: borrowing dai2 does not matter right? we can say just 1 debt and user any other asset (so it's more generic)

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 10 above)
Comment (miguelmtzinf) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: yeah, better to move wbtc utilization up and debt utilization down (or just mock borrow rates)

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 15]
```diff
@@ -0,0 +1,1048 @@
...
+      'Bob user risk premium after collateral accrual'
+    );
+  }
+
+  /// Debt is initially covered by 2 collaterals (dai + dai2), then 1 collateral becomes enough to cover the debt due to interest accrual
+  function test_riskPremium_fuzz_nonIncreasesAfterCollateralAccrual(
+    uint256 daiSupplyAmount,
+    uint40 skipTime
+  ) public {
+    daiSupplyAmount = bound(daiSupplyAmount, 1e18, MAX_SUPPLY_AMOUNT / 2 - 2); // Leave some room to allow borrowing more than this supply, and for Alice borrow
```
Comment (miguelmtzinf) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: wdym with this comment?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 9 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: Do you prefer that @miguelmtzinf ? I can do, but I kinda like showing the strictly increasing case

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 13 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:40]: That's the motivation, yeah. There are other ways to do it, but I have been using this pattern elsewhere in user rp tests, so stuck with it

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 14 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: For sure, I can lean towards these comments being more generic. Just wanted to explicitly describe this test

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 15 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: I need to borrow more dai than this supplied amount as per the test condition to have debt covered by 2 collaterals. I am assuming a borrow cap of `MAX_SUPPLY_AMOUNT / 2`, and I also need Alice to borrow 1 dai to accrue debt interest. That's why there is a `-2`, to not go over the borrow cap

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:1]: Added these

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 10 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: Changed

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 11 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: ty, removed

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 12 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:29]: Done, added a helper function `_calcEquivalentAssetAmount` for this in base

File: tests/Base.t.sol [hunk 16]
```diff
@@ -747,6 +748,27 @@ abstract contract Base is Test {
...
+
+  /// @dev Helper function to calculate the equivalent asset amount for a given asset
+  function _calcEquivalentAssetAmount(
+    uint256 inputAssetId,
+    uint256 inputAssetAmount,
+    uint256 outputAssetId
+  ) internal view returns (uint256) {
+    uint256 valueOfInputAsset = _getValueInBaseCurrency(inputAssetId, inputAssetAmount);
+    uint256 valueOfWeiOutput = _getValueInBaseCurrency(outputAssetId, 1);
+    return valueOfInputAsset / valueOfWeiOutput;
```
Comment (miguelmtzinf) [inline tests/Base.t.sol:775]: this won't work if `valueOfWeiOutput` is > `valueOfInputAsset`
it's an extreme case but worth handling

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 12 above)
Comment (miguelmtzinf) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:29]: shouldn't we use it here to make sure dai or usdx amounts cover the  debt?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 9 above)
Comment (miguelmtzinf) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: probably missing something. how do you make the first being gt and second ge?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 14 above)
Comment (miguelmtzinf) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: it's just that supply and borrow the same asset looks like a very specific case. I d borrow a different asset to make the test more generic

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 9 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: In gt case, I withdraw **all** of the lower lp collateral. In the ge case, I fuzz the withdraw amount, so withdrawing e.g. 1 wei of the asset may not be enough to actually change user rp (due to rounding)

File: tests/Base.t.sol (hunk 16 above)
Comment (yan-man) [reply tests/Base.t.sol:775]: i had to do something similar in liquidations - created the opposite method to convert base currency into amount: https://github.com/aave/aave-v4/blob/eeb781c61a3fe9ba9d6b4972ed007e2c7d50afa1/tests/Base.t.sol#L959

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 17]
```diff
@@ -229,14 +290,39 @@ contract SpokeRiskPremiumEdgeCasesTest is SpokeBase {
...
+      totalDebt: baseDebt,
+      reserveFactor: 0,
+      assetId: wbtcAssetId,
+      virtualUnderlyingBalance: wbtcAsset.availableLiquidity,
+      usingVirtualBalance: true
+    });
+    vm.mockCall(
+      address(irStrategy),
+      abi.encodeWithSelector(IReserveInterestRateStrategy.calculateInterestRates.selector, params),
+      abi.encode(1000e27)
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: would you mind updating to conventions from other places
```suggestion
      abi.encode(uint256(10_00).bpsToRay())
```

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol [hunk 18]
```diff
@@ -0,0 +1,1140 @@
...
+      amount: wbtcSupplyAmount,
+      onBehalfOf: bob
+    });
+
+    // Now risk premium should be less than or equal to LP of weth
+    assertLe(
+      spoke1.getUserRiskPremium(bob),
+      spoke1.getLiquidityPremium(_wethReserveId(spoke1)),
+      'Bob user risk premium after supplying lower LP collateral'
+    );
```
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:759]: wondering if on all these edge case tests whether it's worthwhile also assert the user rp is also eq to `_calculateExpectedUserRP` output, given you already have that helper available, rather than merely directional (Le, Ge, etc)

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol
Comment (yan-man) [inline tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:1]: yeah this test file got quite large, maybe can split into `EdgeCases.unit.t.sol` and `EdgeCases.fuzz.t.sol`?

Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:1]: I just pushed the pattern where I call into the fuzz tests with particular edge case params. It's down to 764 lines. I think it's alright?

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 14 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol]: imho here it's really just meant to be the math that's tested, and it's clean that dai and dai2 have the same decimals. The conversions kinda make the tests messy. And in fact, this test suite is meant to be the edge cases, so I'd also posit that it's appropriate to test this kind of thing here

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 2 above)
Comment (yan-man) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:12]: but on the flip side, if we had 2 colls exactly covering a debt
- collA: 1e5, LP = 10%
- collB: 1 wei, LP = 5%

the user rp would be skewed such that collB should only contribute an amount that rounds to 0 bc it is less than 1 BPS, so that user rp should be 10_00 BPS. But due to precision it'd instead be 99_99 BPS. Is that correct? I ask bc I saw this also in dealing with wAvg collateral factor on HF calcs

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 18 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:759]: Added

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 2 above)
Comment (DhairyaSethi) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:12]: all coll values are normalised to base currency (18 + 8 decimals) tho

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol
Comment (DhairyaSethi) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:1]: agree think it's fine now

File: tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol (hunk 2 above)
Comment (CheyenneAtapour) [reply tests/unit/Spoke/Spoke.RiskPremium.EdgeCases.t.sol:12]: @yan-man I just tested this case. It's correct though that user risk premium should be 9_99 BPS. And it is indeed the case
